editor/editor_settings.cpp

```diff
@@ -1507,7 +1507,13 @@ Ref<Shortcut> ED_GET_SHORTCUT(const String &p_path) {
 		return nullptr;
 	}
 
-	Ref<Shortcut> sc = EditorSettings::get_singleton()->get_shortcut(p_path);
+	Ref<Shortcut> sc;
+#ifdef OSX_ENABLED
+	// See if it has an osx override.
+	sc = EditorSettings::get_singleton()->get_shortcut(p_path + ".osx");
+#endif
```

Comment:
That's a bit hacky, though it's already better than the previous approach.

The clean approach would be to implement the same logic as in `ProjectSettings`:
https://github.com/godotengine/godot/blob/342a0d45cfa0c9cabb76bb6616b0fc2d704766e7/core/config/project_settings.cpp#L184-L202

i.e. actually query OS features to find potential overrides that should be applied.

But if we find that's too complex, the ad hoc hack done here might be sufficient.

Reply:
Ah ok, sorry I didn't really understand the features stuff. I think I do now and will update it tommorow (late here now)